packages/ts/hilla-file-router/src/vite-plugin/collectRoutesFromFS.ts

```diff
@@ -11,11 +11,38 @@ export type RouteMeta = Readonly<{
   children: RouteMeta[];
 }>;
 
+/**
+ * Routes collector options.
+ */
 export type CollectRoutesOptions = Readonly<{
+  /**
+   * The list of extensions for files that will be collected as routes.
+   */
   extensions: readonly string[];
+  /**
+   * The Vite logger instance.
+   */
   parent?: URL;
+  /**
+   * The parent directory of the current directory. This is a
+   * nested parameter used inside the function only.
+   */
+  logger: Logger;
 }>;
 
+async function checkFile(url: URL | undefined, logger: Logger): Promise<URL | undefined> {
+  if (url) {
+    const contents = await readFile(url, 'utf-8');
+    if (contents === '') {
+      return undefined;
+    } else if (!contents.includes('export default')) {
+      logger.error(`The file "${String(url)}" should contain a default export of a component`);
+    }
```

Comment:
Simpler to have only one rule, especially when also taking into account the possibility of having multiple named exports. For what use case would it be useful to support something else?

Reply:
Don't have any specific use case in mind, just wanted to know the "why", and double-check how limiting might this be for developers. In case, it is a common practice e.g. in NextJs, personally have nothing against it.

Reply:
No, named export is not valid. First: it won't work with the TypeScript correctly while we have it imported as a whole module to be used in React Router. Second: it is error-prone because we would need to analyze which named export is exactly the component we want to use. I guess, it is better to use a default export since it is a great tool provided by the ECMA specification

Reply:
An article / documentation will prevent any possible confusions well enough.